lib/pages/sign_up_page.dart

```diff
@@ -0,0 +1,159 @@
+import 'package:economiza_sc/stores/sign_up_store.dart';
+import 'package:economiza_sc/widgets/logo_widget.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_mobx/flutter_mobx.dart';
+import 'package:flutter_modular/flutter_modular.dart';
+
+class SignUpPage extends StatelessWidget {
+  const SignUpPage({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return const Scaffold(
+      body: Center(child: SignUpForm()),
+    );
+  }
+}
+
+class SignUpForm extends StatefulWidget {
+  const SignUpForm({super.key});
+
+  @override
+  State<SignUpForm> createState() => _SignUpFormState();
+}
+
+class _SignUpFormState extends State<SignUpForm> {
+  final _signUpStore = Modular.get<SignUpStore>();
+  final _formKey = GlobalKey<FormState>();
+  final _nameController = TextEditingController();
+  final _emailController = TextEditingController();
+  final _passwordController = TextEditingController();
+  final _passwordConfirmationController = TextEditingController();
+
+  _handleSubmit(BuildContext context) {
+    if (!_formKey.currentState!.validate()) {
+      return;
+    }
+
+    _signUpStore.signUp(
+      _nameController.text,
+      _emailController.text,
+      _passwordController.text,
+      _passwordConfirmationController.text,
+    );
+  }
+
+  bool _isEmail(String input) {
+    final emailRegex = RegExp(r'^[\w-]+(\.[\w-]+)*@([\w-]+\.)+[a-zA-Z]{2,7}$');
+    return emailRegex.hasMatch(input);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Observer(builder: (context) {
+      return ConstrainedBox(
+        constraints: const BoxConstraints(maxWidth: 400),
+        child: Form(
+          key: _formKey,
+          child: Padding(
+            padding: const EdgeInsets.symmetric(vertical: 16, horizontal: 16),
+            child: SingleChildScrollView(
+              child: Column(
+                mainAxisAlignment: MainAxisAlignment.center,
+                children: [
+                  const LogoWidget(assetPath: 'assets/images/logo_full_transparent_bag.png'),
+                  TextFormField(
+                    style: Theme.of(context).textTheme.bodyLarge!.copyWith(color: Colors.black),
+                    key: const Key('nameField'),
+                    controller: _nameController,
+                    decoration: InputDecoration(hintText: 'Nome', hintStyle: Theme.of(context).textTheme.bodyLarge!.copyWith(color: Colors.grey)),
+                    validator: (value) {
```

Comment:
We also have the same validation on the login page. Please, extract the validators in the app so we can reuse them in the future. This can be done on a new PR.